resources/netdata_install.rb

```diff
@@ -30,58 +36,135 @@
+    git new_resource.git_source_directory do
+      repository new_resource.git_repository
+      reference new_resource.git_revision
+      action :sync
+    end
+
+    execute 'install' do
+      cwd new_resource.git_source_directory
+      command <<-EOF
+        ./netdata-installer.sh #{enable_autoupdate} #{custom_install_path} \
+        --zlib-is-really-here --dont-wait --dont-start-it
+      EOF
+      notifies :install, 'package[compile_packages]', :before
+      notifies :restart, 'service[netdata]', :delayed
+      not_if { autoupdate_enabled_on_system? }
+      only_if { update || !netdata_installed? }
+    end
+
+    service 'netdata' do
+      supports status: true, restart: true, reload: true
+      action [:start, :enable]
+    end
+
+  when 'binary'
+    unless %w(amd64 x86_64).include?(node['kernel']['machine'])
+      raise "Static binary versions of netdata are available only for 64bit Intel/AMD CPUs (x86_64), but yours is: #{node['kernel']['machine']}."
+    end
+
+    binary_package = new_resource.binary_package
+    binary_repository = new_resource.binary_repository
+
+    if binary_package == 'netdata-latest.gz.run'
+      remote_file "#{Chef::Config[:file_cache_path]}/netdata_latest.txt" do
+        source "#{binary_repository}/#{binary_package}"
+        action :create
+      end
+      ruby_block 'set_netdata_binary_package_env' do
+        block do
+          ENV['NETDATA_BINARY_PACKAGE'] = ::File.read("#{Chef::Config[:file_cache_path]}/netdata_latest.txt")
+        end
+      end
+      remote_file "#{Chef::Config[:file_cache_path]}/netdata-package.gz.run" do
+        source lazy { "#{binary_repository}/#{::File.read("#{::Chef::Config[:file_cache_path]}/netdata_latest.txt")}" }
+        action :create
+      end
+      file "#{Chef::Config[:file_cache_path]}/netdata_latest.txt" do
+        action :delete
+      end
+    else
+      ruby_block 'set_netdata_binary_package_env' do
+        block do
+          ENV['NETDATA_BINARY_PACKAGE'] = binary_package
+        end
+      end
+      remote_file "#{Chef::Config[:file_cache_path]}/#{binary_package}" do
+        source "#{binary_repository}/#{binary_package}"
+        action :create
+      end
+    end
+
+    unless new_resource.binary_install_options.include?('--accept')
+      Chef::Log.warn 'It is required to provide --accept option to the binary package installation script to accept the license' \
+                     'otherwise automated installation will fail.'
+    end
+
+    binary_install_options = new_resource.binary_install_options.join(' ')
+    binary_post_install_options =
+      new_resource.binary_post_install_options.empty? ? ' ' : '--' + ' ' + new_resource.binary_post_install_options.join(' ')
+
+    bash 'install_netdata_binary_package' do
+      code <<-EOH
+        sh #{Chef::Config[:file_cache_path]}/netdata-package.gz.run #{binary_install_options} #{binary_post_install_options}
+      EOH
+      not_if { ::File.exist?('/opt/netdata/bin/netdata') }
```

Comment:
If we skip installation if the directory already exist we could also avoid downloading the binary package as well.

Reply:
true